Update docs for required argument of `find`

Minirogue opened the pull request:
`find()` has a required argument of type `KClass`, so `find<User>()` doesn't work since it has no arguments.

An alternative solution, of course, would be to change the `find()` function to match the docs if that is the desired behavior. I'm not sure if the `type` argument is used in a way that can't be done with just the generic `M`. If not, then the argument can just be removed in favor of the generic type. Otherwise I don't know an immediate solution offhand.

SalomonBrys commented:
Thanks !

SalomonBrys commented:
Thanks !
We updated the documentation to move the cursor documentation in its own page.
Can you update your PR?

Minirogue commented on README.md:
```diff
@@ -32,21 +32,21 @@ db.put(User("John", "Doe"))
-If you feel like enhancing this document, you can propose a pull request that modifies https://github.com/Kodein-Framework/Kodein-DB/tree/master/doc[the documentation documents].
-(Documentation is auto-generated from those).
+If you feel like enhancing this document, you can propose a pull request that modifies [the documentation documents](https://github.com/Kodein-Framework/Kodein-DB/tree/master/doc).
+The generated documentation can be found [here](https://docs.kodein.org/kodein-db/0.3.0/index.html).
```
Is there a way to link to the latest docs? Or will this have to be updated with every release?

Minirogue commented:
Moved changes to `iteration.adoc`.
Also fixed the links in `README.md` to what I believe was intended as well as fixed the example to what should be the correct usage.

SalomonBrys commented:
Sorry, but I just realized the `find<Type>()` function does exist, but needs to be imported:

https://github.com/Kodein-Framework/Kodein-DB/blob/master/kdb/kodein-db-api/src/commonMain/kotlin/org/kodein/db/DBRead.kt#L29
